source/tool/tests/software/mock_helper.py

```diff
@@ -130,6 +130,55 @@ def read_phys_mem(self, pa_hi, pa_lo, length):
         else:
             return "\xFF" * length
 
+class DSDTParsingHelper(ACPIHelper):
+    """Test helper containing generic descriptor for RSDP, RSDT, and FADT to parse DSDT
+
+    Default entry for DSDT and X_DSDT inside FADT is 0x0
+
+    One additional region is defined:
+      * FADT table [0x400, 0x514]
+    """
+    USE_FADT_WITH_X_DSDT = True
+
+    FADT_ADDRESS = 0x400
+    DSDT_ADDRESS = 0x0
+    X_DSDT_ADDRESS = 0x0
+
+    def _ensure_fadt_in_sdt(self, SDT_ENTRIES):
+        if self.FADT_ADDRESS not in SDT_ENTRIES:
+            SDT_ENTRIES.append(self.FADT_ADDRESS)
```

Comment:
Done, but not sure whether it's good enough. I avoided using `hasattr` because it doesn't look good :s